crates/ironrdp-cliprdr/src/pdu/format_list.rs

```diff
@@ -13,6 +13,15 @@ pub struct ClipboardFormat {
     pub name: String,
 }
 
+impl ClipboardFormat {
+    pub fn new_standard(id: u32) -> Self {
```

Comment:
Consider making a `ClipboardFormatId` enum so that it's clear what this id means.

Reply:
I thought about that but rejected the idea because clipboard IDs could be actually arbitrary, OS-specific, enums are not ideal for such things, I think.

Maybe the right solution would be the new type struct `struct ClipboardFromatId(pub u32)`
and then define constants inside its `impl`:

```rust
impl ClipboardFromatId {
    const CF_TEXT: Self = Self(1);
    const CF_UNICODE_TEXT: Self = Self(13);
    // ...
}
```

This should make usage more clear:

```rust
let format_id = ClipboardFromatId::CF_TEXT;

// some custom id
let format_id = ClipboardFromatId(42);
```

I think it will be good compromise between raw `u32` and too constrained `enum`

Reply:
> Maybe the right solution would be the new type struct `struct ClipboardFromatId(pub u32)` and then define constants inside its impl

Yes, this is generally a good approach in network protocols for parsing resilience. :+1: 

There is already some occurrence for this pattern:
- [`Code` in `ironrdp_graphics::rle` module](https://github.com/Devolutions/IronRDP/blob/55176ff99b8c72076c49ec4a2333ba9324798bc5/crates/ironrdp-graphics/src/rle.rs#L386-L446)
- [`FailureCode` in `ironrdp_pdu::nego` module](https://github.com/Devolutions/IronRDP/blob/55176ff99b8c72076c49ec4a2333ba9324798bc5/crates/ironrdp-pdu/src/nego.rs#L61-L85)
- [SecurityType in IronVNC](https://github.com/Devolutions/IronVNC/blob/547b482b2b516d176a80a5bdf09e1d4dae2878bc/crates/pdu/src/handshake.rs#L607-L708) (repository is private)

That’s the way to go when knowledge of _ALL_ the values is not required, not desired or not possible and we need to handle gracefully these situations.